torch/csrc/jit/serialization/import_source.cpp

```diff
@@ -408,6 +417,25 @@ struct SourceImporterImpl : public Resolver,
                 for (const auto& buffer : buffer_list) {
                   buffer_names.insert(StringLiteral(buffer).text());
                 }
+              } else if (name == "__forward_pre_hooks__") {
+                TORCH_INTERNAL_ASSERT(
+                    is_module,
+                    "Forward pre hooks only exist on modules at the moment");
+                const auto pre_hook_list =
+                    ListLiteral(assign.rhs().get()).inputs();
+                for (const auto& pre_hook : pre_hook_list) {
+                  pre_hook_names.insert(StringLiteral(pre_hook).text());
+                  pre_hooks_order.emplace_back(StringLiteral(pre_hook).text());
+                }
+              } else if (name == "__forward_hooks__") {
+                TORCH_INTERNAL_ASSERT(
+                    is_module,
+                    "Forward hooks only exist on modules at the moment");
+                const auto hook_list = ListLiteral(assign.rhs().get()).inputs();
+                for (const auto& hook : hook_list) {
+                  hook_names.insert(StringLiteral(hook).text());
+                  hooks_order.emplace_back(StringLiteral(hook).text());
+                }
```

Comment:
This will likely pose forward compatibility problems. I think we need to gate this with `#ifndef FBCODE_CAFFE2` and then remove the guard in a few months time.

cc: @gmagogsfm, exact same problem that caused the SEV we discussed.

Reply:
What issue is this gate trying to solve/prevent? The TK_DEF case below on 473 will treat the pre_hooks as methods if they're not added to the list here. I might have to gate things in python_print.cpp as well

Reply:
I added the \#ifndef to the part where the hooks are printed into the IR, so this code won't be triggered. Would that take care of the forward compatibility issue?

Reply:
Yes - and you're right, we don't need to add the macro here, only n `python_print.cpp`.